bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/network/SiemensHvacRequestListener.java

```diff
@@ -0,0 +1,253 @@
+import org.eclipse.jetty.client.api.Response.ContentListener;
+import org.eclipse.jetty.client.api.Response.FailureListener;
+import org.eclipse.jetty.client.api.Response.SuccessListener;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.client.util.BufferingResponseListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+public class SiemensHvacRequestListener extends BufferingResponseListener
+        implements SuccessListener, FailureListener, ContentListener, CompleteListener, QueuedListener, BeginListener {
+
+    public enum ErrorSource {
+        ErrorBridge,
+        ErrorThings
+    }
+
+    private static int onSuccessCount = 0;
+    private static int onBeginCount = 0;
+    private static int onQueuedCount = 0;
+    private static int onCompleteCount = 0;
+    private static int onFailureCount = 0;
+
+    private final Logger logger = LoggerFactory.getLogger(SiemensHvacRequestListener.class);
+
+    private SiemensHvacRequestHandler requestHandler;
+    private SiemensHvacConnector hvacConnector;
+
+    /**
+     * Callback to execute on complete response
+     */
+    private final SiemensHvacCallback callback;
+
+    public static int getQueuedCount() {
+        return onQueuedCount;
+    }
+
+    public static int getStartedCount() {
+        return onBeginCount;
+    }
+
+    public static int getCompleteCount() {
+        return onCompleteCount;
+    }
+
+    public static int getFailureCount() {
+        return onFailureCount;
+    }
+
+    public static int getSuccessCount() {
+        return onSuccessCount;
+    }
+
+    /**
+     * Constructor
+     *
+     * @param callback Callback which execute method has to be called.
+     */
+    public SiemensHvacRequestListener(SiemensHvacRequestHandler requestHandler) {
+        this.requestHandler = requestHandler;
+        this.hvacConnector = requestHandler.getHvacConnector();
+        this.callback = requestHandler.getCallback();
+    }
+
+    @Override
+    public void onSuccess(@Nullable Response response) {
+        onSuccessCount++;
+        requestHandler.setResponse(response);
+
+        if (response != null) {
+            logger.debug("{} response: {}", response.getRequest().getURI(), response.getStatus());
+        }
+    }
+
+    @Override
+    public void onFailure(@Nullable Response response, @Nullable Throwable failure) {
+        onFailureCount++;
+        requestHandler.setResponse(response);
+
+        if (response != null && failure != null) {
+            Throwable cause = failure.getCause();
+            if (cause == null) {
+                cause = failure;
+            }
+
+            String msg = cause.getLocalizedMessage();
+
+            if (cause.getClass() == ConnectException.class) {
```

Comment:
Perhaps this would be a bit more clear?
```suggestion
            if (cause instanceof ConnectException) {
```

(and of course for the others below as well)